python/paddle/fluid/dygraph/dygraph_to_static/ifelse_transformer.py

```diff
@@ -130,7 +145,10 @@ def is_candidate_node(node):
     """
     Nodes with specified type will be dependent on tensor.
     """
-    return isinstance(node, (gast.Compare, gast.BoolOp, gast.UnaryOp))
+    is_compare_node = isinstance(node,
+                                 (gast.Compare, gast.BoolOp, gast.UnaryOp))
+    has_numpy_attr = ".numpy()" in ast_to_source_code(node)
```

Comment:
This can be a TODO thing, nothing is required now. 

We used some strings in ProgramTranslator to classify usage, but users may create a class with same string name. We had better to find a better way in the future.

Reply:
You are right, this is hard code and we should deal with this case more elegent.